src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -123,25 +125,41 @@ public bool StartAssignment(HostAssignmentContext context)
             }
         }
 
-        public async Task<string> ValidateContext(HostAssignmentContext assignmentContext)
+        public Task<string> ValidateContext(HostAssignmentContext assignmentContext)
         {
             _logger.LogInformation($"Validating host assignment context (SiteId: {assignmentContext.SiteId}, SiteName: '{assignmentContext.SiteName}')");
+            RunFromPackageContext pkgContext = assignmentContext.GetRunFromPkgContext();
+            _logger.LogInformation($"Will be using {pkgContext.EnvironmentVariableName} app setting as zip url");
 
+            if (pkgContext.IsScmRunFromPackage())
+            {
+                // Not user assigned so limit validation
+                return null;
```

Comment:
This needs to be `Task.FromResult<string>(null)`. Otherwise, it throws `Object reference not set to an instance of an object.`, because `await null` will trigger an error.

Reply:
Thank you for catching that.